README.md

```diff
@@ -77,22 +89,132 @@ In the sub-folders for plots that intersected with the orthomosaic, there will b
+docker volume create my_output
+``` 
+
+Step 2 involves copying the drone images into a folder:
+```bash
+mkdir -p /inputs
+cp /IMG/* /inputs/
+```
+
+Step 3 copies the optional shapefile files into the same folder as the drone images:
+```bash
+cp /plot_shapes.* /inputs/
+``` 
+
+In step 4 we copy the experiment.yaml file into the same folder as the drone images:
+```bash
+cp experiment.yaml /inputs/
+``` 
+
+In step 5 we copy the source files onto the input named volume:
+```bash
+docker run --rm -v /inputs:/sources -v my_input:/input --entrypoint bash agdrone/canopycover-workflow:latest -c 'cp /sources/* /input/'
+``` 
+
+In step 6 we create a local folder to hold the output from processing:
+```bash
+mkdir -p /output
+```
+
+In step 7 we run the workflow to generate the orothomosaic image using ODM (OrthoDroneMap) and calculate plot-level canopy cover:
+```bash
+docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v /inputs:/scif/data/odm/images -v scif_output:/output -e INPUT_VOLUME=my_input -e OUTPUT_VOLUME=my_output -e "INPUT_IMAGE_FOLDER=/images" -e "OUTPUT_FOLDER=/output" agdrone/canopycover-workflow:latest run odm_workflow plot_shapes.shp my_input my_output
+```
+and we wait until it's finished.
+
+In step 8 we copy the results off the named output volume to our local folder:
+```bash
+docker run --rm -v /output:/results -v my_output:/output --entrypoint bash agdrone/canopycover-workflow:latest -c 'cp -r /output/* /results/'
+```
+The results of the processing are now in the `/output` folder.
+
+Finally, in step 9 we clean up the named volumes by deleting everything on them:
+```bash
+docker run --rm -v my_input:/input -v my_output:/output --entrypoint bash agdrone/canopycover-workflow:latest -c 'rm -r /input/* && rm -r /output/*'
+```
+
 ### Clean
+
 By executing the [scif](#scif) app named `clean` it's possible to clean up the output folder and other generated files.
 It's recommended, but not necessary, to run the clean app between processing runs by either running this command or through other means.
 
 **Example:**
 
-This docker command line will clean up the output files generated using the [Canopy Cover: Orthomosaic and Shapefile](#can_om_shp) example above.
+This docker command line will clean up the output files generated using the [Canopy Cover: Orthomosaic and Shapefile](#om_can_shp) example above.
 ```bash
-docker run --rm -v ~/inputs:/scif/data/odm/images -v ~/outputs:/scif/data/soilmask agdrone/canopycover-shape-workflow:latest run clean
+docker run --rm -v /inputs:/scif/data/odm/images -v /outputs:/scif/data/soilmask agdrone/canopycover-shape-workflow:latest run clean
 ```
 
 ## Build the container
+
 This section describes how the Docker container could be built.
 Please refer to the [Docker](https://www.docker.com/) documentation for more information on building Docker containers.
 
 ```bash
 cp jx-args.json.example jx-args.json
 docker build --progress=plain -t agdrone/canopycover-shape-workflow:latest .
```

Comment:
This `docker build` command will result in an error unless the user enabled Docker BuildKit (which there is no instruction for). Either remove `--progress=plain`,  or add the line to enable Docker BuildKit if it's required.

(Also see earlier comment about image name. Is it `agdrone/canopycover-shape-workflow` or `agdrone/canopycover-workflow`?)

Reply:
Furthermore I can't get this container to build, neither on my Mac nor on Linux.

I get this error:

```
ERROR: Could not find a version that satisfies the requirement pycparser (from cffi>=1.4.1->cryptography->agpypeline) (from versions: none)
ERROR: No matching distribution found for pycparser (from cffi>=1.4.1->cryptography->agpypeline)
ERROR Return value 256 for install of soilmask
Installing base at /scif
+ apprun     soilmask
+ apphelp     soilmask
+ appinstall soilmask
The command '/bin/sh -c scif install /opt/soilmask_v0.0.1_ubuntu16.04.scif' returned a non-zero code: 1
```

I'm not sure, but I wonder if the dependency resolution problem has something to do with using an old version of `agpypeline` (0.0.20 as opposed to 0.0.41).

As a result I can't test this PR further.

Reply:
It would be good to have automatic continuous integration tests with Docker builds set up to catch issues like this on new PRs, as well scheduled (nightly?) builds for main branches to catch when silent updates to upstream dependencies break something.

Reply:
I was just thinking that about CI and was adding a ticket to that effect!

Reply:
FYI, `--progress=plain` is still there as I mentioned earlier.

Reply:
Still can't build the image:

```
Installing collected packages: agpypeline
Successfully installed agpypeline-0.0.37
Traceback (most recent call last):
  File "src/soilmask.py", line 10, in <module>
    import cv2
  File "/scif/apps/soilmask/.venv/lib/python3.8/site-packages/cv2/__init__.py", line 5, in <module>
    from .cv2 import *
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
ERROR Return value 256 for install of soilmask
Installing base at /scif
+ apprun     soilmask
+ apphelp     soilmask
+ appinstall soilmask
The command '/bin/sh -c scif install /opt/soilmask_v0.0.1_ubuntu16.04.scif' returned a non-zero code: 1
```

Reply:
Maybe this is related? https://github.com/planetlabs/notebooks/pull/113/files

Reply:
This just popped up for me as well. The build now works